src/Reflection/ReflectionClass.php

```diff
@@ -698,23 +714,36 @@ private function createImmediateConstants(ClassNode|InterfaceNode|TraitNode|Enum
      */
     public function getConstants(int $filter = 0): array
     {
+        return $this->getConstantsHelper($filter);
+    }
+
+    /**
+     * @param int-mask-of<ReflectionClassConstantAdapter::IS_*> $filter
+     * @param list<class-string>                                $classNameStack
+     *
+     * @return array<non-empty-string, ReflectionClassConstant> indexed by name
+     */
+    private function getConstantsHelper(int $filter = 0, array &$classNameStack = []): array
```

Comment:
`Helper` doesn't say anything: can this be renamed? What does it **do**? :P

Reply:
`doGetConstants()` would already be better than `*Helper` BTW

Reply:
Perhaps `getConstantsConsideringAlreadyVisitedClasses()`?

Reply:
yeah, it did hurt a bit typing out "helper" tbh :D